### PR TITLE
[ Microsoft.SCVMM ] Requesting vmmserver credentials until non-empty

### PR DESCRIPTION
**Pull request by nascarsayan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
connectedvmware

File: src/scvmm/HISTORY.rst [hunk 1]
```diff
@@ -26,3 +26,8 @@ Release History
 * Force and retain flags in VM delete
 * Generated SDK now requires updated version of azure-cli
 * Long running PATCH operations
+
+0.1.5
++++
+* Requesting VMMServer credentials from the user until they are empty.
```
Comment (wangzelin007) [inline src/scvmm/HISTORY.rst:32]:
```suggestion
* Requesting VMMServer credentials from the user until they are non-empty.
```

**Comment by nascarsayan** [comment]
@zhoxing-ms , could you merge the PR if everything looks fine?

File: src/scvmm/HISTORY.rst [hunk 2]
```diff
@@ -26,3 +26,8 @@ Release History
 * Force and retain flags in VM delete
 * Generated SDK now requires updated version of azure-cli
 * Long running PATCH operations
+
+0.1.5
++++
+* Requesting VMMServer credentials from the user until they are non-empty.
+* Removing default value for port. Asking for the input. If input is empty, setting port to 8100.
```
Comment (zhoxing-ms) [inline src/scvmm/HISTORY.rst:33]: Removing the default value will cause breaking change, right? If so, please add `[BREAKING CHANGE]` as the prefix
```suggestion
* [BREAKING CHANGE] Removing default value for port. Asking for the input. If input is empty, setting port to 8100.
```

Comment (zhoxing-ms) [inline src/scvmm/HISTORY.rst:33]: In addition, the history notes of new version should be written at the top, history notes should be arranged from new to old from top to bottom

File: src/scvmm/azext_scvmm/custom.py [hunk 3]
```diff
@@ -88,26 +88,43 @@ def connect_vmmserver(
...
+        while not creds['port']:
+            print('Please provide vmmserver port (Default: 8100): ', end='')
+            try:
+                creds['port'] = input()
+                if not creds['port']:
+                    creds['port'] = DEFAULT_VMMSERVER_PORT
+                creds['port'] = int(creds['port'])
+            except ValueError:
+                print('Port must be a number, please try again')
+                creds['port'] = None
```
Comment (zhoxing-ms) [inline src/scvmm/azext_scvmm/custom.py:118]: For the automation scenario, if the CLI script does not pass in the `creds['port']` parameter, the process of CLI script will get stuck. Is this in line with expectations?

Comment (nascarsayan) [reply src/scvmm/azext_scvmm/custom.py:118]: Yes, this is in line with the expectations. We will add modify the command by adding the new param to our automation pipelines.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**